### PR TITLE
Updates functools.py with consistent quotes

### PR DESCRIPTION
**Pull request by sobolevn**
I have noticed that `'` quotes are used everywhere except this particular case,
which was introduced in https://github.com/python/cpython/pull/18726

So, this is a trivial fix to enforce better consistency.

Automerge-Triggered-By: @csabella

### CODE REVIEW

**Comment by the-knights-who-say-ni** [comment]
Hello, and thanks for your contribution!

I'm a bot set up to make sure that the project can legally accept this contribution by verifying everyone involved has signed the [PSF contributor agreement](https://www.python.org/psf/contrib/contrib-form/) (CLA).

## Recognized GitHub username

We couldn't find a [bugs.python.org](https://bugs.python.org/) (b.p.o) account corresponding to the following GitHub usernames:

@sobolevn

This might be simply due to a missing "GitHub Name" entry in one's b.p.o account settings. This is necessary for legal reasons before we can look at this contribution. Please follow [the steps outlined in the CPython devguide](https://devguide.python.org/pullrequest/#licensing) to rectify this issue.




You can [check yourself](https://check-python-cla.herokuapp.com/) to see if the CLA has been received.

Thanks again for the contribution, we look forward to reviewing it!

**Comment by sobolevn** [comment]
I have signed CLA, but it says that I have to wait at least one US business day. Ok, let's wait 🙂

**Review by aeros — COMMENTED** [review]
Welcome @sobolevn!

Although I'll admit that it looks slightly odd to use both single quote and double quote strings in the same list, we typically discourage non-functional changes in the source code (not including large organizational refactors or optimizations). This helps to minimize code churn, makes the revision history easier to read, and maximizes the limited (entirely voluntary) review time. While this one alone may not seem like an issue, it's more about the precedent when you consider the volume of PRs we receive. 

If you're looking to make a first contribution, I'd highly recommend checking out the issues on bugs.python.org, particularly ones with the keyword "newcomer friendly" or "easy" [1]. We also welcome improvements to the documentation, including the addition of docstrings to functions/methods that didn't previously have one, comment fixing, and various other improvements to the docs on docs.python.org. Typically, similarly minimal documentation changes are more likely to be merged because they pose less possible downsides and risk. As an example, here was my first merged PR: https://github.com/python/cpython/pull/14069.

That being said, I'll leave it to the discretion of the maintainer for the functools module, @rhettinger, to decide the best course of action for the PR.

---

[1] - For convenience, here's a couple of links:
[All open "newcomer friendly" issues on bpo](https://bugs.python.org/issue?%40search_text=&ignore=file%3Acontent&title=&%40columns=title&id=&%40columns=id&stage=&creation=&creator=&activity=&%40columns=activity&%40sort=activity&actor=&nosy=&type=&components=&versions=&dependencies=&assignee=&keywords=22&priority=&status=1&%40columns=status&resolution=&nosy_count=&message_count=&%40group=&%40pagesize=50&%40startwith=0&%40sortdir=on&%40queryname=&%40old-queryname=&%40action=search).
[All open "easy" issues on bpo](https://bugs.python.org/issue?%40search_text=&ignore=file%3Acontent&title=&%40columns=title&id=&%40columns=id&stage=&creation=&creator=&activity=&%40columns=activity&%40sort=activity&actor=&nosy=&type=&components=&versions=&dependencies=&assignee=&keywords=6&priority=&status=1&%40columns=status&resolution=&nosy_count=&message_count=&%40group=&%40pagesize=50&%40startwith=0&%40sortdir=on&%40queryname=&%40old-queryname=&%40action=search).

If you'd like to tackle any of the above issues or have any questions, feel free to add me to the "nosy list" of the issue or @ mention me in the PR. :)

Also, I'd recommend checking out the following resource if you haven't already: https://devguide.python.org/#contributing

**Review by rhettinger — APPROVED** [review]

**Comment by rhettinger** [comment]
Am marking this as approved.  Please get the CLA signed.

Also, we don't typically accept minor "consistency" patches and instead defer to existing code, so please don't use this acceptance as an indication to make more patches like this.  Instead, please look to the tracker for some real bugs to fix :-)

**Comment by aeros** [comment]
@sobolevn Regarding the CLA status, you may have to manually update it through the [CLA assistant heroku app](https://check-python-cla.herokuapp.com/). If it shows that your CLA has been signed, @bedevere-bot will swap the label from `CLA not signed` to `CLA signed`.

**Comment by sobolevn** [comment]
Done!

Thanks a lot!

**Pull request merged**
